File: solutions/013.Roman_to_Integer/AC_simulation_1.java
/*
*  Author:      illuz <iilluzen[at]gmail.com>
*  File:        AC_simulation_1.java
*  Create Date: 2015-03-02 23:14:38
*  Descripton:   
*/

import java.util.Scanner;

public class Solution {
    private int[] val = new int[255];
    private void init() {
        val['I'] = 1; val['V'] = 5; val['X'] = 10; val['L'] = 50;
        val['C'] = 100; val['D'] = 500; val['M'] = 1000;
    }

    public int romanToInt(String s) {
        init();
        int ret = 0;
        for (int i = 0; i < s.length(); i++) {
            if (i > 0 && val[s.charAt(i)] > val[s.charAt(i - 1)]) {
                ret += val[s.charAt(i)] - 2 * val[s.charAt(i - 1)];
            } else {
                ret += val[s.charAt(i)];
            }
        }
        return ret;
    }

    // debug
    public static void main(String[] args) {
        Scanner cin = new Scanner(System.in);
        Solution s = new Solution();
        int[] input = {1, 2, 3, 1};
        System.out.println("no case");
    }
}

File: solutions/013.Roman_to_Integer/AC_simulation_1.cpp
/*
*  Author:      illuz <iilluzen[at]gmail.com>
*  File:        AC_simulation_1.cpp
*  Create Date: 2015-01-08 14:47:56
*  Descripton:  simulation
*/

#include <bits/stdc++.h>

using namespace std;
const int N = 0;

class Solution {
private:
    int val[255];
    void init() {
        val['I'] = 1; val['V'] = 5; val['X'] = 10; val['L'] = 50;
        val['C'] = 100; val['D'] = 500; val['M'] = 1000;
    }
public:
    int romanToInt(string s) {
        init();
        int ret = 0;
        for (int i = 0; i < s.size(); i++) {
            if (i > 0 && val[s[i]] > val[s[i - 1]]) {
                ret += val[s[i]] - 2 * val[s[i - 1]];
            } else {
                ret += val[s[i]];
            }
        }
        return ret;
    }
};

int main() {
    string str;
    Solution s;
    while (cin >> str)
        cout << s.romanToInt(str) << endl;
    return 0;
}

